src/main/java/ay2122s1_cs2103t_w16_2/btbb/model/shared/GenericStringPredicate.java

```diff
@@ -0,0 +1,40 @@
+package ay2122s1_cs2103t_w16_2.btbb.model.shared;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/**
+ * Tests that a generic string matches any of the keywords given.
+ *
+ * @param <T> Type of the predicate.
+ */
+public class GenericStringPredicate<T> implements Predicate<T> {
```

Comment:
I think it is better not to associate this with `GenericString` because we have a class called `GenericString`, but this can be used for classes other than `GenericString`.

Or do you have a better naming idea?
```suggestion
public class StringContainsKeywordPredicate<T> implements Predicate<T> {
```